src/Hangfire.Mongo/Database/HangfireDbContext.cs

```diff
@@ -39,8 +46,56 @@ public HangfireDbContext(IMongoClient mongoClient, string databaseName, string p
             Client = mongoClient;
             Database = mongoClient.GetDatabase(databaseName);
             ConnectionId = Guid.NewGuid().ToString();
+
+            GetServerVersion();
+        }
+
+        /// <summary>
+        /// Try to get server version, sometimes it can be useful to use certain feature only
+        /// if the version is greater than a minimum version.
+        /// </summary>
+        private void GetServerVersion()
+        {
+            //We cache version in memory, it is unlikely that the database changes version while the application is running and
+            //if we want to use newer feature it is a good idea to restart.
+            if (_version != null)
+            {
+                return;
+            }
+            try
+            {
+                //We need to get server version to check if we can user various features.
+                var adminDatabase = Client.GetDatabase("admin");
+
+                // Check the server version
+                var buildInfoCommand = new BsonDocumentCommand<BsonDocument>(new BsonDocument { { "buildInfo", 1 } });
+                var buildInfo = adminDatabase.RunCommand(buildInfoCommand);
+                var versionString = buildInfo.GetValue("version").AsString;
+                _version = Version.Parse(versionString);
+                if (Logger.IsInfoEnabled())
+                {
+                    Logger.InfoFormat("Mongodb version is {0}", _version);
+                }
+            }
+            catch (Exception e)
+            {
+                //we are unable to determine version of the server, this is a warning, now knowing a version will not cause

```

Comment:
typo: "we are unable to determine MongoDB Server version, this is a warning; not knowing..."